serverv2/appmanager/app_manager_full.go

```diff
@@ -0,0 +1,119 @@
+package appmanager
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+)
+
+type Mempool interface {
+}
+
+type AppManagerBuilder struct {
+	InitGenesis map[string]func(ctx context.Context, moduleGenesisBytes []byte) error
+}
+
+func (a *AppManagerBuilder) RegisterInitGenesis(moduleName string, genesisFunc func(ctx context.Context, moduleGenesisBytes []byte) error) {
+	a.InitGenesis[moduleName] = genesisFunc
+}
+
+func (a *AppManagerBuilder) RegisterHandler(moduleName, handlerName string, handler MsgHandler) {
+	panic("...")
+}
+
+type MsgSetKVPairs struct {
+	Pairs []ChangeSet
+}
+
+func (a *AppManagerBuilder) Build() *AppManager {
+	genesis := func(ctx context.Context, genesisBytes []byte) error {
+		genesisMap := map[string][]byte{} // module=> genesis bytes
+		for module, genesisFunc := range a.InitGenesis {
+			err := genesisFunc(ctx, genesisMap[module])
+			if err != nil {
+				return fmt.Errorf("failed to init genesis on module: %s", module)
+			}
+		}
+		return nil
+	}
+	return &AppManager{initGenesis: genesis}
+}
+
+type AppManager struct {
+	// configs
+	checkTxGasLimit uint64
+	queryGasLimit   uint64
+	// configs - end
+
+	db Store
+
+	lastBlockHeight *atomic.Uint64
+
+	initGenesis func(ctx context.Context, genesisBytes []byte) error
+
+	stf *STFAppManager
+
```

Comment:
should we have the grpc routers at this level to avoid the dependency in the stf builder and router?